source/ethereum-clients/cpp-ethereum/building-from-source/linux.rst

```diff
@@ -5,20 +5,56 @@ Building for Linux
 ##################
 
 
-Clone the repository
-====================
+Getting the source code
+===================
 
-To clone the source code, execute the following commands:
+We use git and GitHub to maintain the source code. Clone the repository by:
 
 .. code:: bash
 
     git clone --recursive https://github.com/ethereum/cpp-ethereum.git
     cd cpp-ethereum
+    
+The ``--recursive`` option is important. It orders git to clone also submodules which are required to build the project.
+If you missed it you can correct your mistake with command ``git submodule update --init``.
+
+
+CMake
+=====
+
+We use CMake to control the build configuration of the project. Quite recent version of CMake is required (at the time of writing 3.4 is the minimum). We recomand to install the latest version downloading and unpacking the binary distribution of the latest version awailable on the CMake download page:
```

Comment:
Please break long lines. Also: Shouldn't it be "recommend installing" instead of "recommend to install"? Not 100% sure, though.